executor/cook/progress.py

```diff
@@ -256,14 +257,19 @@ def __update_progress(self, progress_report):
                          'progress-sequence': self.sequence_counter.increment_and_get()}
         return True
 
-    def retrieve_progress_states(self):
+    def retrieve_progress_states(self, os_error_handler):
```

Comment:
Why not rethrow the OSError and handle it in the caller instead of passing a handler?

Reply:
Done.